### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vsupruniuk**

### CODE REVIEW

**Comment by vsupruniuk** [comment]
image
image
image
image
image
image
image
image
image

File: templates/taxi/car_confirm_delete.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{% extends "base.html" %}
+
+{% block content %}
+    <h1>Delete Car</h1>
+
+    <p>Are you sure you want to delete the car: {{ car }}</p>
+
+    <form action="" method="POST">
+        {% csrf_token %}
+        <input class="btn btn-danger" type="submit" value="Confirm"/>
```
Comment (LiudmylaKulinchenko) [inline templates/taxi/car_confirm_delete.html:10]: Add Cancel button apart from Delete one. The Cancel button will lead to the previous page the user was on -- checklist

File: templates/taxi/manufacturer_confirm_delete.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <h1>Delete Manufacturer</h1>
+
+    <p>Are you sure you want to delete the manufacturer: {{ manufacturer }}</p>
+
+    <p>All manufacturer cars also will be deleted</p>
+
+    <form action="" method="POST">
+        {% csrf_token %}
+        <input class="btn btn-danger" type="submit" value="Confirm"/>
+    </form>
```
Comment (LiudmylaKulinchenko) [inline templates/taxi/manufacturer_confirm_delete.html:13]: Add Cancel button apart from Delete one

File: taxi/views.py [hunk 3]
```diff
@@ -34,6 +36,25 @@ class ManufacturerListView(LoginRequiredMixin, generic.ListView):
...
+    form_class = ManufacturerForm
+    success_url = reverse_lazy("taxi:manufacturer-list")
+    template_name = "taxi/manufacturer_form.html"
+
+
+class ManufacturerDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Manufacturer
+    success_url = reverse_lazy("taxi:manufacturer-list")
+    template_name = "taxi/manufacturer_confirm_delete.html"
+
```
Comment (LiudmylaKulinchenko) [inline taxi/views.py:56]: Don't override template_name, context_object_name and so on if they are the same as default ones -- checklist
Same for car views

File: taxi/urls.py [hunk 4]
```diff
@@ -16,8 +22,26 @@
...
+    path(
+        "manufacturers/<int:pk>/delete",
+        ManufacturerDeleteView.as_view(),
+        name="manufacturer-delete"
+    ),
     path("cars/", CarListView.as_view(), name="car-list"),
     path("cars/<int:pk>/", CarDetailView.as_view(), name="car-detail"),
+    path("cars/create", CarCreateView.as_view(), name="car-create"),
+    path("cars/<int:pk>/update", CarUpdateView.as_view(), name="car-update"),
+    path("cars/<int:pk>/delete", CarDeleteView.as_view(), name="car-delete"),
```
Comment (LiudmylaKulinchenko) [inline taxi/urls.py:44]: Missing a trailing `/` in the URL

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

**Review by LiudmylaKulinchenko — APPROVED** [review]
Nice work! 💪

**Pull request closed without merging**